decompiler/pipeline/ssa/phi_lifting.py

```diff
@@ -56,6 +54,8 @@ def _lift_phi_functions_of(self, basic_block: BasicBlock) -> None:
                 new_basic_block = self._insert_basic_block_before(basic_block, new_instructions)
                 if predecessor:
                     self._cfg.substitute_edge(edge, edge.copy(sink=new_basic_block))
+                else:
+                    self._cfg.root = self._cfg.get_roots()[0]
```

Comment:
But why not setting it directly to the new root `new_basic_block`? The given basic-block has no predecessor, so it must be the root. Hence, the new root must be the new constructed basic-block.